### PR TITLE
detect/ipopts: Multiple option support

### PR DESCRIPTION
**Pull request by jlucovsky**
Continuation of #10688 

Support multiple options.

This PR changes the IP option definitions from an enum into bit values so the values added during packet parsing are compared properly when evaluation with an IP option specified with `ipopts` occurs.

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: [6864](https://redmine.openinfosecfoundation.org/issues/6864)

Describe changes:
- Misc. cleanups
- Move IPv4 option values to a bit mask
- suricata-verify test to validate each option lacking coverage.

Updates:
- Remove unneeded PCRE usage

### Provide values to any of the below to override the defaults.

To use a pull request use a branch name like `pr/N` where `N` is the
pull request number.

Alternatively, `SV_BRANCH` may also be a link to an
OISF/suricata-verify pull-request.

```
SV_REPO=
SV_BRANCH=pr/1722
SU_REPO=
SU_BRANCH=
LIBHTP_REPO=
LIBHTP_BRANCH=
```

### CODE REVIEW

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 19751

File: src/detect-ipopts.c [hunk 1]
```diff
@@ -173,11 +162,7 @@ static int DetectIpOptsMatch (DetectEngineThreadCtx *det_ctx, Packet *p,
     if (!de || !PKT_IS_IPV4(p) || PKT_IS_PSEUDOPKT(p))
         return 0;
 
-    if (p->ip4vars.opts_set & de->ipopt) {
-        return 1;
-    }
-
-    return 0;
+    return p->ip4vars.opts_set & de->ipopt;
```
Comment (catenacyber) [inline src/detect-ipopts.c:165]: I still think the Match functions are documented to return 1 or 0, not any positive integer

File: src/detect-ipopts.c [hunk 2]
```diff
@@ -32,10 +32,6 @@
 #include "detect-ipopts.h"
 #include "util-unittest.h"
 
-#define PARSE_REGEX "\\S[A-z]"
-
-static DetectParseRegex parse_regex;
```
Comment (catenacyber) [inline src/detect-ipopts.c:43]: This PCRE removal could be its own commit ;-)

Comment (jlucovsky) [reply src/detect-ipopts.c:43]: I'm including it as part of the overhaul to make this detection function properly.

**Comment by jlucovsky** [comment]
Continued in #10741
